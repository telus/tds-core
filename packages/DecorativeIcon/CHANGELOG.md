# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.1.0](https://github.com/telusdigital/tds/compare/@tds/core-decorative-icon@1.0.2...@tds/core-decorative-icon@1.1.0) (2019-03-15)


### Features

* **core-decorative-icon:** add accessible svg icon ([402e9b7](https://github.com/telusdigital/tds/commit/402e9b7))





## [1.0.2](https://github.com/telusdigital/tds/compare/@tds/core-decorative-icon@1.0.1...@tds/core-decorative-icon@1.0.2) (2019-03-05)

**Note:** Version bump only for package @tds/core-decorative-icon





## [1.0.1](https://github.com/telusdigital/tds/compare/@tds/core-decorative-icon@1.0.0...@tds/core-decorative-icon@1.0.1) (2019-02-06)

**Note:** Version bump only for package @tds/core-decorative-icon
