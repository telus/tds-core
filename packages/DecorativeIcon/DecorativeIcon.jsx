import React from 'react'
import PropTypes from 'prop-types'

import Icon from '../../shared/components/Icon/Icon'

/**
 * An icon used for visual aesthetics only.
 */
const DecorativeIcon = ({ symbol, variant, size, ...rest }) => (
  <Icon {...rest} symbol={symbol} variant={variant} size={size} aria-hidden="true" />
)

DecorativeIcon.propTypes = {
  /**
   * Name of the icon symbol.
   */
  symbol: PropTypes.oneOf([
    'caretDown',
    'caretUp',
    'checkmark',
    'chevron',
    'leftChevron',
    'exclamationPointCircle',
    'expander',
    'hamburger',
    'location',
    'minus',
    'plus',
    'questionMarkCircle',
    'spyglass',
    'times',
  ]).isRequired,
  /**
   * The appearance of the Icon.
   */
  variant: PropTypes.oneOf(['primary', 'secondary', 'inverted', 'error']),
  /**
   * The icon size in pixels.
   */
  size: PropTypes.oneOf([16, 24, 48]),
}

DecorativeIcon.defaultProps = {
  variant: undefined,
  size: 24,
}

export default DecorativeIcon
