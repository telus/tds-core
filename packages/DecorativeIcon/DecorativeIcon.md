Decorative icons do not perform a role beyond visual aesthetics and are hidden from screen readers using the aria-hidden attribute. Usually, the information being communicated with the icon is also conveyed in another manner.
