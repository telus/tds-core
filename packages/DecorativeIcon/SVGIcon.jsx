import React from 'react'
import PropTypes from 'prop-types'

import styled from 'styled-components'
import { colorTelusPurple, colorShark, colorWhite } from '@tds/core-colours'

import safeRest from '../../shared/utils/safeRest'

const getColour = variant => {
  switch (variant) {
    case 'alternative':
      return colorShark
    case 'inverted':
      return colorWhite
    case 'default':
    default:
      return colorTelusPurple
  }
}

const svgVariant = ({ variant }) => ({ '& > svg': { fill: getColour(variant) } })
const svgSize = ({ size }) => ({ '& > svg': { width: size, height: size } })

const StyledSVGIcon = styled.i(
  {
    display: 'inline-flex',
  },
  svgVariant,
  svgSize
)
/**
 * SVG Icons are a type of DecorativeIcon that use SVGs instead of fonts to render the icons.
 *
 * @version ./package.json
 */
const SVGIcon = props => {
  return <StyledSVGIcon {...safeRest(props)} aria-hidden="true" />
}

SVGIcon.propTypes = {
  /**
   * @ignore
   */
  children: PropTypes.node.isRequired,
  /**
   * The appearance of the Icon.
   */
  variant: PropTypes.oneOf(['default', 'alternative', 'inverted']),
  /**
   * The icon size in pixels.
   */
  size: PropTypes.oneOf([16, 24, 48]),
}

SVGIcon.defaultProps = {
  variant: 'default',
  size: 24,
}

export default SVGIcon
