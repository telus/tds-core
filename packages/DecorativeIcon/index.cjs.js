const DecorativeIcon = require('./dist/index.cjs')

module.exports = DecorativeIcon
