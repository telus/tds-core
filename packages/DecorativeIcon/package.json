{
  "name": "@tds/core-decorative-icon",
  "version": "2.18.1",
  "description": "An icon used for visual aesthetics only",
  "main": "index.cjs.js",
  "module": "index.es.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/telusdigital/tds.git"
  },
  "publishConfig": {
    "access": "public"
  },
  "author": "TELUS digital",
  "engines": {
    "node": ">=8"
  },
  "bugs": {
    "url": "https://github.com/telusdigital/tds/issues"
  },
  "homepage": "http://tds.telus.com",
  "peerDependencies": {
    "react": "^16.8.2 || ^17.0.0",
    "react-dom": "^16.8.2 || ^17.0.0",
    "styled-components": "^4.1.3 || ^5.1.0"
  },
  "dependencies": {
    "@tds/core-colours": "^2.2.2",
    "@tds/util-helpers": "^1.6.1",
    "prop-types": "^15.5.10"
  }
}
