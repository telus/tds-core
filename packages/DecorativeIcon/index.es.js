import './dist/index.css'
import DecorativeIcon from './dist/index.es'

export default DecorativeIcon
export * from './svgs'
