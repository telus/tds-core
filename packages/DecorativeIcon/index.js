import DecorativeIcon from './DecorativeIcon'

export default DecorativeIcon
export * from './svgs'
