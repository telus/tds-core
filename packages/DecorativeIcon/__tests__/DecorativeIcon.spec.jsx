import React from 'react'
import { shallow, render } from 'enzyme'

import DecorativeIcon from '../DecorativeIcon'

describe('DecorativeIcon', () => {
  const defaultProps = {
    symbol: 'spyglass',
  }
  const doShallow = (props = {}) => shallow(<DecorativeIcon {...defaultProps} {...props} />)

  it('renders', () => {
    const icon = render(<DecorativeIcon symbol="spyglass" />)

    expect(icon).toMatchSnapshot()
  })

  it('passes attributes to the Icon component', () => {
    const icon = doShallow({ symbol: 'checkmark', variant: 'secondary', size: 16, id: 'the-icon' })

    expect(icon).toHaveProp('symbol', 'checkmark')
    expect(icon).toHaveProp('variant', 'secondary')
    expect(icon).toHaveProp('size', 16)
    expect(icon).toHaveProp('id', 'the-icon')
  })

  it('is hidden from screen readers', () => {
    const icon = doShallow()

    expect(icon).toHaveProp('aria-hidden', 'true')
  })
})
