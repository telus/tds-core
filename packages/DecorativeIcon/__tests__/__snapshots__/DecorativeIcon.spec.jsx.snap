// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DecorativeIcon renders 1`] = `
.c0 {
  font-family: 'TELUS Core Icons';
  display: inline-block;
  font-weight: normal;
  font-style: normal;
  speak: none;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  text-transform: none;
  text-rendering: auto;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  line-height: 1;
  vertical-align: middle;
  font-size: 1.5rem;
}

.c0::before {
  content: '\\f111';
}

<i
  aria-hidden="true"
  class="c0"
/>
`;
